server.js

```diff
@@ -12,23 +13,50 @@ function CouchUrlRewriteProxy (opts) {
       headers: req.headers,
       qs: req.query
     }
+    var pipe
     req.headers.host = 'registry.npmjs.org'
 
     if (~['PUT', 'POST', 'DELETE'].indexOf(req.method)) payload.body = req.body
 
-    request(payload, function (err, response, body) {
+    if (
+      !req.path.match(/\/-\//) && // CouchDB API URLs.
+      !req.path.match(/\.tgz$/) && // tarball URLs.
+      req.method === 'GET' // we should only rewrite GET requests!
+    ) {
+      pipe = false
```

Comment:
Might be easier to read/grok if this var was named `rewrite` and its value inverted.